src/main/java/uk/ac/cam/cl/dtg/isaac/api/managers/EventBookingManager.java

```diff
@@ -392,6 +399,101 @@ public EventBookingDTO requestBooking(final IsaacEventPageDTO event, final Regis
+        for (RegisteredUserDTO user : users) {
+            this.ensureValidBooking(event, user, true);
+        }
+
+        List<EventBookingDTO> reservations = new ArrayList<>();
+        try {
+            // Obtain an exclusive database lock to lock the event
+            this.bookingPersistenceManager.acquireDistributedLock(event.getId());
+
+            // is there space on the event? Teachers don't count for student events.
+            // work out capacity information for the event at this moment in time.
+            // If there is no space, no reservations are made. Throw an exception and handle in EventsFacade.
+            this.ensureCapacity(event, users);
+
+            // IMPORTANT: Any non-ignorable exception past this point must roll back any reservation made thus far.
+            for (RegisteredUserDTO user : users) {
+                // attempt to book them on the event
+                EventBookingDTO reservation;
+
+                // attempt to book them on the event
+                if (this.hasBookingWithStatus(event.getId(), user.getId(), BookingStatus.CANCELLED)) {
+                    // if the user has previously cancelled we should let them book again.
+                    reservation = this.bookingPersistenceManager.updateBookingStatus(event.getId(), user.getId(), reservingUser.getId(),
+                            BookingStatus.RESERVED, null);
+                } else {
+                    reservation = this.bookingPersistenceManager.createBooking(event.getId(), user.getId(), reservingUser.getId(),
+                            BookingStatus.RESERVED, null);
+                }
+                reservations.add(reservation);
+
+                try {
+                    // af599 TODO: Use the correct email template here.
+                    // af599 TODO: Content team action.
+                    emailManager.sendTemplatedEmailToUser(user,
+                            emailManager.getEmailTemplateDTO("email-event-booking-confirmed"),
+                            new ImmutableMap.Builder<String, Object>()
+                                    .put("contactUsURL", generateEventContactUsURL(event))
+                                    .put("authorizationLink", String.format("https://%s/account?authToken=%s",
+                                            propertiesLoader.getProperty(HOST_NAME), event.getIsaacGroupToken()))
+                                    .put("event.emailEventDetails", event.getEmailEventDetails() == null ? "" : event.getEmailEventDetails())
+                                    .put("event", event)
+                                    .build(),
+                            EmailType.SYSTEM,
+                            Collections.singletonList(generateEventICSFile(event, reservation)));
+
+                } catch (SegueDatabaseException | ContentManagerException e) {
+                    log.error(String.format("Unable to send welcome email (%s) to user (%s)",
```

Comment:
might want to update from welcome email to event email - I suspect there are a number of places in the file with this copy and paste error (many of them my fault)